sceptre/hooks/__init__.py

```diff
@@ -84,7 +57,7 @@ def __set__(self, instance: "Stack", value):
         """
 
         def setup(attr, key, value: Hook):
-            attr[key] = clone = value.clone(instance)
+            attr[key] = clone = value.clone_for_stack(instance)
```

Comment:
`clone_for_stack` (as you'll see below) is a recursive clone operation that allows for cloning resolvers inside of the arguments.